jdaviz/configs/imviz/plugins/orientation/orientation.py

```diff
@@ -351,36 +354,54 @@ def add_orientation(self, rotation_angle=None, east_left=None, label=None,
         )
 
         # add orientation layer to all viewers:
-        self._add_data_to_all_viewers(label)
+        for viewer_ref in self.app._viewer_store:
+            self._add_data_to_viewer(label, viewer_ref)
 
         if set_on_create:
-            self.orientation._update_layer_items()
             self.orientation.selected = label
 
-    def _add_data_to_all_viewers(self, data_label):
-        for viewer_ref in self.app.get_viewer_reference_names():
-            layers = [
-                layer.label for layer in
-                self.app.get_viewer(viewer_ref).layers
-            ]
-            if data_label not in layers:
-                self.app.add_data_to_viewer(viewer_ref, data_label)
+    def _add_data_to_viewer(self, data_label, viewer_id):
+        viewer = self.app.get_viewer_by_id(viewer_id)
+
+        wcs_only_layers = viewer.state.wcs_only_layers
+        if data_label not in wcs_only_layers:
+            self.app.add_data_to_viewer(viewer_id, data_label)
 
     def _on_viewer_added(self, msg):
-        for data_label in self.orientation.choices:
-            self._add_data_to_all_viewers(data_label)
+        self._send_wcs_layers_to_all_viewers(viewers_to_update=[msg._viewer_id])
+
+    @observe('viewer_items')
+    def _send_wcs_layers_to_all_viewers(self, *args, **kwargs):
+        if not hasattr(self, 'viewer'):
+            return
+
+        wcs_only_layers = self.app.get_viewer_by_id('imviz-0').state.wcs_only_layers
+
+        viewers_to_update = kwargs.get(
+            'viewers_to_update', self.app.get_viewer_reference_names()
+        )
 
-        if hasattr(self, 'orientation') and len(self.orientation.choices):
-            self.viewer.selected = msg._viewer_id
-            self.orientation.selected = self.orientation.choices[0]
+        for viewer_ref in viewers_to_update:
+            for wcs_layer in wcs_only_layers:
+                if wcs_layer not in self.app.get_viewer_by_id(viewer_ref).state.wcs_only_layers:
+                    self.app.add_data_to_viewer(viewer_ref, wcs_layer)
+
+            self.viewer.selected = viewer_ref
+
+            self.orientation.set_wcs_only_filter(wcs_only=self.link_type_selected == 'WCS')
+            # select the default orientation if no data have yet been
+            # added to a newly created viewer:
+            if (
+                    base_wcs_layer_label in self.orientation.choices and
+                    not len(self.viewer.selected_obj.layers)
+            ):
+                self.orientation.selected = base_wcs_layer_label
 
     def _on_data_add_to_viewer(self, msg):
-        if (
-            msg._viewer_reference != 'imviz-0' and
-            self.app.get_viewer_by_id(msg._viewer_reference).state.reference_data is None
-        ):
-            self.viewer.selected = msg._viewer_reference
-            self.orientation.selected = base_wcs_layer_label
+        if msg._viewer_id != 'imviz-0':

```

Comment:
Again, should we use `default_image_viewer` to avoid the hardcoded `imviz-0`?

But also, would this ever be needed if `imviz-0` is emptied by the user first?

Reply:
Since the user isn't allowed to delete `imviz-0`, this probably isn't necessary, I wrote this conditional as part of logic that was unnecessary, so I can remove it now.